osu.Game/Online/API/Requests/GetScoresRequest.cs

```diff
@@ -27,12 +31,67 @@ public GetScoresRequest(BeatmapInfo beatmap)
             Success += onSuccess;
         }
 
+        public GetScoresRequest(BeatmapInfo beatmap, RulesetInfo ruleset, LeaderboardScope scope = LeaderboardScope.Global)
+        {
+            if (!beatmap.OnlineBeatmapID.HasValue)
+                throw new InvalidOperationException($"Cannot lookup a beatmap's scores without having a populated {nameof(BeatmapInfo.OnlineBeatmapID)}.");
+
+            this.beatmap = beatmap;
+            this.scope = scope;
+            this.ruleset = ruleset;
+
+            Success += onSuccess;
+        }
+
         private void onSuccess(GetScoresResponse r)
         {
             foreach (OnlineScore score in r.Scores)
                 score.ApplyBeatmap(beatmap);
         }
 
+        protected override WebRequest CreateWebRequest()
+        {
+            var req = base.CreateWebRequest();
+
+            switch(scope)
+            {
+                default:
+                case LeaderboardScope.Global:
+                    req.AddParameter(@"type", @"global");
+                    break;
+
+                case LeaderboardScope.Friends:
+                    req.AddParameter(@"type", @"friend");
+                    break;
+
+                case LeaderboardScope.Country:
+                    req.AddParameter(@"type", @"country");
+                    break;
+            }
+
+            switch (ruleset?.Name)

```

Comment:
we never want these constants to exist in the code. either the api should be changed to accept the ID rather than a name, or we should store these names as an `OnlineIdentifier` or something inside `RulesetInfo` instead.

Reply:
yeah it felt wrong but was the only way that i could see to get it working immediately. which would be the preferred solution?

Reply:
The API should be changed to support IDs.

Reply:
Tracking at https://github.com/ppy/osu-web/issues/1819